src/main/scala/ai/privado/languageEngine/java/passes/config/PropertiesFilePass.scala

```diff
@@ -35,19 +34,23 @@ import overflowdb.BatchedUpdate
 import overflowdb.traversal._
 
 import scala.jdk.CollectionConverters._
-import java.util.Properties
+import java.util.{Properties}
 import scala.util.{Failure, Success, Try}
 import io.shiftleft.semanticcpg.language._
 import io.circe.yaml.parser
 import com.github.wnameless.json.flattener.JsonFlattener
 
+import scala.xml._
+
 /** This pass creates a graph layer for Java `.properties` files.
   */
 class PropertiesFilePass(cpg: Cpg, projectRoot: String) extends ForkJoinParallelCpgPass[String](cpg) {
 
   private val logger = LoggerFactory.getLogger(getClass)
 
-  override def generateParts(): Array[String] = propertiesFiles(projectRoot).toArray
+  var testCpg: Cpg = _
+  override def generateParts(): Array[String] =
+    propertiesFiles(projectRoot, Set(".properties", ".yml", ".yaml", ".xml")).toArray
```

Comment:
Are we filtering the files with .properties and other extensions?

Reply:
Yes, we need files from the project only with the selected extensions.